components/search_bar/search_bar.jsx

```diff
@@ -81,15 +81,16 @@ export default class SearchBar extends React.PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
-        const {searchTerms} = nextProps;
+    static getDerivedState(props, state) {
+        const {searchTerms} = props;
 
-        this.setState({
-            visibleSearchHintOptions: this.determineVisibleSearchHintOptions(searchTerms),
-        });
+        return {
+            ...state,
+            visibleSearchHintOptions: SearchBar.determineVisibleSearchHintOptions(searchTerms)
+        }
```

Comment:
Heads up that CI reports two linter errors: /root/mattermost/mattermost-webapp/components/search_bar/search_bar.jsx
  89:95  error  Missing trailing comma  comma-dangle
  90:10  error  Missing semicolon       semi

Reply:
Ok i just did a commit that fixes the problems